src/uu/touch/src/touch.rs

```diff
@@ -333,7 +354,17 @@ const YYYYMMDDHHMM_OFFSET_FORMAT: &[time::format_description::FormatItem] = form
     [offset_hour sign:mandatory][offset_minute]"
 );
 
-fn parse_date(s: &str) -> UResult<FileTime> {
+fn filetime_to_offset_dt(ft: FileTime) -> OffsetDateTime {
+    OffsetDateTime::from_unix_timestamp(ft.unix_seconds())
+        .unwrap()
+        .checked_add(Duration::nanoseconds(ft.nanoseconds().into()))
+        .unwrap()
+}
+
+fn parse_date(
```

Comment:
This function is getting complex enough to need some documentation. Could you add a docstring explaining what it does, which formats it accepts and what the arguments and resulted values are?

Reply:
I'm also wondering if this is the right approach, maybe it makes sense to decouple the parsing and the reference times? If `parse_date` had a signature like this it might be easier:
```rust
// TODO: needs a better name :)
enum Date {
    Absolute(FileTime),
    Relative(Duration),
}

fn parse_date(s: &str) -> Date { ... }
```
So that the calling function could then apply the right reference times?